lib/store/index.js

```diff
@@ -34,29 +40,74 @@ class Store {
   }
 
   @action
-  newKernel(kernel: Kernel) {
-    const mappedLanguage =
-      Config.getJson("languageMappings")[kernel.language] || kernel.language;
-    this.runningKernels.set(mappedLanguage, kernel);
+  newKernel(
+    kernel: Kernel,
+    filePath: string,
+    editor: atom$TextEditor,
+    grammar: atom$Grammar
+  ) {
+    if (isMultilanguageGrammar(editor.getGrammar())) {
+      this.kernelMapping.merge({ [filePath]: { [grammar.name]: kernel } });
```

Comment:
I tested this out in a multi-kernel .md file and had some issues. `this.kernelMapping.merge` is overwriting the filename key instead of adding a grammar key (unless that is intended?).

Reply:
Good catch! I reverted this change since `.merge()` doesn't recursively merge the object.